lib/loader/ast/type.cpp

```diff
@@ -9,22 +9,21 @@ namespace WasmEdge {
 namespace Loader {
 
 // Load binary decode HeapType. See "include/loader/loader.h".
-Expect<RefType> Loader::loadHeapType(RefTypeCode Code, ASTNodeAttr From) {
+Expect<HeapType> Loader::loadHeapType(ASTNodeAttr From) {
```

Comment:
As mentioned above, your `HeapType` implementation here will always create a value type with `RefNull`.
It losts the information of the reference type. The value type may be `Ref` or `RefNull`.